src/docs/stan-reference/appendices.tex

```diff
@@ -645,15 +645,16 @@ \subsection{Variable Declarations}
 
 range_constraint ::= ?('<' range '>')
 
-range ::= 'lower' '=' expression ',' 'upper' = expression
-        | 'lower' '=' expression
-        | 'upper' '=' expression
+range ::= 'lower' '=' range_expression ',' 'upper' = range_expression
```

Comment:
See below---the current `range_expression` isn't right and I'd suggest just treating the value as a semantic constraint rather than trying to pack it into the syntax.

Reply:
Fair enough.  I wrote it out that way because it was the easiest way to get the GLR parser that tree-sitter uses to correctly parse the range constraints and it seemed like the Spirit parser implements it as a separate grammar rather than just a semantic constraint.

Reply:
I originally missed that the last option in the range grammar was to call the common grammar.  But I thought the range expressions were more restrictive.

This works:

```
real<lower = exp(-1.9), upper = 2.9 + 1.3> x;
```
but doesn't seem to be allowed in the specific range expression grammar.   

But you're right that it's not a free-for-all. I had forgotten that this isn't allowed (until reminded by your next comment):

```
  real<lower = 1 > 2> y;
```
I gave up before figuring out how to do the backtracking for this case.

Reply:
I think it does. Since `range_expression` includes the`common_expression` expression, it also includes `identifier '(' expression ')'`, which would handle the lower bound,`exp(-1.9)`.  The `range_expression` rule includes the expression `range_expression arthmeticInfixOp range_expression` which would match `2.9 + 1.3` in the upper bound.   

Handling range expressions was the main/only reason that I separated the infix operators into logical (`||`, `&&`, `<`, `<=`, `>=`, `>`) operators (not allowed in range expressions) and arithmetic (the other higher precedence operators allowed in range expressions). From scanning the code and empirical trial and error, this is how I understood how it is being handled by the Stan parser.

Reply:
And even though it's a "hack", there's really no expressivity lost since this equivalent way of writing that variable declaration is valid
```
real<lower = (1 > 2)> y;
```